tests/auto/rt_auto.py

```diff
@@ -0,0 +1,474 @@
+
+    def get_yaml_subset(self, keyin):
+        self.logger.info(f'Getting YAML subset: {keyin}')
+        try:
+            yaml_subset = self.data[keyin]
+        except:
+            self.logger.critical(f'Unable to get yaml subset: {keyin}. Quitting')
+            sys.exit()
+        df = pd.DataFrame.from_dict(yaml_subset)
+        self.logger.info(f'Finished getting YAML subset {keyin}')
+        return df
+
+class Machine:
+
+    def __init__(self, rtdata_obj):
+        self.logger = logging.getLogger("Machine")
+        self.rtdata_obj = rtdata_obj
+        self.get_machine_info()
+        self.machineid = os.environ.get('MACHINE_ID')
+
+    def get_machine_info(self):
+        self.logger.info(f'Getting machine information')
+        hostname = socket.gethostname()
+        self.logger.debug(f'hostname is {hostname}')
+        machines = self.rtdata_obj.get_yaml_subset('machines')
+        regexmachines = machines['regexhostname'].values
+        for i, mach in enumerate(regexmachines):
+            if re.search(mach, hostname):
+                self.logger.info(f'{hostname} is an approved machine')
+                mach_db_info = machines.iloc[i]
+                self.name = mach_db_info['name']
+                self.regexhostname = mach_db_info['regexhostname']
+                self.workdir = mach_db_info['workdir']
+                self.baselinedir = mach_db_info['baselinedir']
+                self.logger.debug(f'self.name: {self.name}\n\
+                    self.regexhostname: {self.regexhostname}\n\
+                    self.workdir: {self.workdir}\n\
+                    self.baselinedir: {self.baselinedir}')
+                self.logger.info(f'Finished getting machine information')
+                return
+            else:
+                continue
+        self.logger.critical(f'Hostname:{hostname} does not match approved list. Quitting')
+        sys.exit()
+
+class Function:
+
+    def __init__(self, name, command, callback):
+        self.logger = logging.getLogger("Function")
+        self.name = name
+        self.command = command
+        self.callback = callback
+        self.logger.debug(f'Function object created')
+        self.logger.debug(f'self.name: {self.name}\n\
+            self.command: {self.command}\n\
+            self.callback: {self.callback}')
+
+    def verify_name(self, comparable):
+        self.logger.debug(f'Verifying name: {comparable}')
+        if re.match(self.name.lower(), comparable.lower()):
+            self.logger.debug(f'Match found')
+            return True
+        else:
+            self.logger.debug(f'Not a match')
+            return False
+
+    def verify_command(self, comparable):
+        self.logger.debug(f'Verifying command: {comparable}')
+        if re.match(self.command.lower(), comparable.lower()):
+            self.logger.debug(f'Match found')
+            return True
+        else:
+            self.logger.debug(f'Not a match')
+            return False
```

Comment:
What is the difference between these two?
The code construct is the exact same.  The comments are different.
This is repetition.